src/validation.cpp

```diff
@@ -905,11 +909,11 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
         assert(ws.m_iters_conflicting.size() == 1);
         CTxMemPool::txiter conflict = *ws.m_iters_conflicting.begin();
 
-        m_limits.descendant_count += 1;
-        m_limits.descendant_size_vbytes += conflict->GetSizeWithDescendants();
+        maybe_rbf_limits.descendant_count += 1;
+        maybe_rbf_limits.descendant_size_vbytes += conflict->GetSizeWithDescendants();
```

Comment:
This prevents `PackageMempoolChecks` and `SubmitPackage` from seeing the bumped limits. Maybe have `const m_base_limits` and `m_current_limits` and have `PreChecks()` reset `m_current_limits=m_base_limits` before bumping it?